CurrentChargingTest.cpp

```diff
@@ -0,0 +1,20 @@
+#include "CurrentCharging.h"
+#include <cassert>
+
+int main()
+{
+    vector<int> periodicSamples = {1, 3, 5, 3, 4, 2, 27, 29, 28, 30, 10, 11, 12, 53, 51, 52};
+    CCurrentReport obj(periodicSamples);
+    obj.generateContinuesCurrentRange();
+    obj.consoleReport();
+    assert(obj.validateRangeandReadings(1, 5, 6) == true);
+    assert(obj.validateRangeandReadings(10, 12, 3) == true);
+    assert(obj.validateRangeandReadings(27, 30, 4) == true);
+    assert(obj.validateRangeandReadings(51, 53, 3) == true);
+    assert(obj.validateRangeandReadings(50, 53, 3) == false);
```

Comment:
Why is a single number not a range? Is this desired functionality or your assumption?
E.g., if many readings show 53, then that's the most frequent reading. I would like to see that fact and not disregard 53 because there is no 52 or 54.
Always question the purpose of the asserts while writing them.